src/tf_branch_deploy/lifecycle.py

```
@@ -0,0 +1,207 @@
+    def remove_non_sticky_lock(self, environment: str) -> None:
+        """Remove lock if it's not sticky."""
+        lock_ref = f"{environment}-branch-deploy-lock"
+        console.print("   🔓 Checking for non-sticky lock")
+
+        # 1. Get lock content
+        try:
+            content_json = self._gh_api_get_content(
+                f"repos/{self.repo}/contents/lock.json", ref=lock_ref
+            )
+            sticky = content_json.get("sticky", "false")
+        except Exception:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)